src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -41,6 +40,13 @@ internal class FindUsagesLSPContext : FindUsagesContext
         private readonly Dictionary<DefinitionItem, int> _definitionToId =
             new Dictionary<DefinitionItem, int>();
 
+        /// <summary>
+        /// Keeps track of definitions that cannot be reported without references and which we have
+        /// not yet found a reference for.
+        /// </summary>
+        private readonly List<VSReferenceItem> _definitionsWithoutReferences =
+            new List<VSReferenceItem>();
```

Comment:
we have a problem here in that if FAR finishes and finds *no* references to *anything*, we still will never report these.  FAR in VS hanldes this specially and still gives an error message.  What does LSP show in this case?

Note: this is not a huge issue, but we may need to do something later.

Reply:
By "no references to anything", are you referring to the case when not even the definition itself is included in the results? It looks like LSP doesn't show anything in that case. I don't know if displaying an error message is something that can be done on our side, but I can file a bug on the LSP side.
LSP did add a progress indicator for FAR recently, so I think that will at least help a bit.